frontend/src/components/applyForm/widgets/WidgetRenderers.tsx

```diff
@@ -50,6 +50,9 @@ export const renderWidget = ({
     console.error(`Unknown widget type: ${type}`, definition);
     throw new Error(`Unknown widget type: ${type}`);
   }
+  if (props.isFormLocked) {
```

Comment:
thanks for making this change - with this in place I think you can roll back the `readOnly` changes to each of the individual widgets other than the ones that weren't already handling disabled field states (which I think is the attachment widgets and budget form widgets)